src/engines/fx/pingpong/pingpong.cpp

```diff
@@ -0,0 +1,81 @@
+#include "pingpong.hpp"
+
+#include "core/service.hpp"
+#include "services/application.hpp"
+#include "services/audio_manager.hpp"
+#include "services/clock_manager.hpp"
+#include "services/log_manager.hpp"
+#include "services/ui_manager.hpp"
+#include "util/math.hpp"
+
+namespace otto::engines::pingpong {
+
+  PingPong::PingPong()
+    : audio(std::make_unique<Audio>()), screen_(std::make_unique<Screen>()), props{{*audio, *screen_}}
+  {}
+
+  using namespace core::input;
+
+  bool PingPong::keypress(Key key)
+  {
+    switch (key) {
+      case Key::blue_click: {
+        props.timetype = !props.timetype;
+        props.delaytime.set(calculate_delaytime(props.timetype, props.free_time, props.subdivision));
```

Comment:
This should probably be done in an on_change handler on timetype

Reply:
Done